packages/react-aria-components/src/ListBox.tsx

```diff
@@ -230,6 +235,21 @@ function ListBoxInner<T extends object>({state: inputState, props, listBoxRef}:
     );
   }
 
+  let sentinelRef = useRef(null);
+  // TODO: Should scrollOffset for useLoadMore should be configurable by the user
+  let memoedLoadMoreProps = useMemo(() => ({
+    isLoading: props.isLoading,
+    onLoadMore: props.onLoadMore,
+    collection,
+    sentinelRef
+  }), [props.isLoading, props.onLoadMore, collection]);
+  // TODO: maybe this should be called at the ListBox level and the StandaloneListBox level. At its current place it is only called
+  // when the Listbox in the dropdown is rendered. The benefit to this would be that useLoadMore would trigger a load for the user before the
+  // dropdown opens but the current state gives the user more freedom as to whether they would like to pre-fetch or not
+  useLoadMore(memoedLoadMoreProps, listBoxRef);
+
+  // TODO: Think about if completely empty state. Do we leave it up to the user to setup the two states for empty and empty + loading?
+  // Do we add a data attibute/prop/renderprop to ListBox for isLoading
```

Comment:
Keep both, will be clearer with refactor to loadingSentinel

Reply:
At the moment, `renderEmptyState` is meant for rendering the empty state AND the initial load spinner, whereas the LoadingSentinel is more for the "loadingMore" spinner